miso-web/src/main/webapp/scripts/shared_hot.js

```diff
@@ -478,5 +478,127 @@ var Hot = {
     } else {
       document.getElementById('saveSuccesses').classList.add('hidden');
     }
+  },
+  _newData : [],
+  incrementingAutofill : function(start, end, rows) {
+    var incrementingContentColumns = {};
+    var valid = true;
+    (function() {
+      // validation check
+      console.log(rows);
+      console.log(start.row);
+      var cols = rows[start.row - 1];
+      if (!cols) {
+        valid = false;
+        return;
+      }
+      for (var cols_i = 0; cols_i < cols.length; cols_i++) {
+        if (!cols[cols_i]) {
+          valid = false;
+          return;
+        }
+      }
+    })();
+    if (!valid) {
+      // allow default functionality to take over.
+      return;
+    }
+    (function() {
+      var cols = rows[0];
+      for (var cols_i = 0; cols_i < cols.length; cols_i++) {
+        var cellContents = "" + cols[cols_i];
+        var numbersInCell = cellContents.match(/\d+/g);
+        var template = cellContents;
+        if (!numbersInCell) {
+          break;
+        }
+        for (var placeHolder_i = 0; placeHolder_i < numbersInCell.length; placeHolder_i++) {
+          template = template.replace(numbersInCell[placeHolder_i], "{"
```

Comment:
The regex is designed to be iterative with /g, so it will return an array of matches.
The placeholder value is based on the array index of each value so you don't get repetition.